File: tests/e2e/if_else.py
# RUN: uv run xdsl-exo -o %t.mlir %s
# RUN: mlir-opt --convert-arith-to-llvm --convert-cf-to-llvm --convert-func-to-llvm --reconcile-unrealized-casts %t.mlir | mlir-translate --mlir-to-llvmir | llc -filetype=obj -o %t.o
# RUN: clang %t.o %S/if_else.c -o %t.bin -lm
# RUN: %t.bin | filecheck %s

from __future__ import annotations

from exo import *


# CHECK: 1.0
# CHECK: 2.0
@proc
def if_else(out: f32[1] @ DRAM, a: index, b: index):
    assert a >= 0
    assert b >= 0
    if a < b:
        out[0] = 1.0
    else:
        out[0] = 2.0

File: tests/e2e/if_else.c
#include <stdio.h>

extern void if_else(float *out, long a, long b);

int main() {
    float out = 0.0f;

    if_else(&out, 1, 5);
    printf("%.1f\n", out);

    if_else(&out, 5, 1);
    printf("%.1f\n", out);

    return 0;
}
